### ISSUE TITLE
Remove all growls on NewDot

### ISSUE THREAD

**Issue by conorpendergrast** [Expensify/App#13229]
On hold for WAQ focus

As identified in [this issue](https://github.com/Expensify/App/issues/13209) and discussed in this thread, we have some remaining growls that are now out of style. This is not a bug, but still needs to be fixed. @luacmartins found the following that still have growls: 

1. Interacting with the checkbox in Workspace > Manage member page
3. Adding/Deleting PaypalMe address
4. Failure updating the User avatar
5. Failure to create, update or delete a Workspace
6. Error downloading a wallet statement

And we should remove them all.

**Comment by trjExpensify** [comment]
>Opening external links when offline

I believe these growls were intentionally left during the Offline API refactors project. The notion being to :do-nothing: as the redirects will go away as we bring those features into NewDot as we progress with reunification. CC: @robertjchen

**Comment by conorpendergrast** [comment]
Ok, removed from the list!

**Comment by kadiealexander** [comment]
Assigning this to myself, as part of my mentorship this year is to be working towards fixing E/app issues. If this becomes an urgent priority please let me know.

**Comment by JmillsExpensify** [comment]
Nice, I'm glad we have a project for this. For what it's worth, I would also remove this one from the list. It's ridiculously far out and something we fix post-reunification. 

> Error downloading a wallet statement

**Comment by kadiealexander** [comment]
Not overdue, have been OOO from a family emergency. Going to dig into this over the next week.

**Comment by JmillsExpensify** [comment]
Nice, this is a pretty good polish project that's worth picking up now. Happy to help with any questions.

**Comment by kadiealexander** [comment]
Thanks Jason! I have a [draft PR](https://github.com/Expensify/App/pull/14405) up with the first growl removed, going to plug away at the rest this week! Should I still hold off on "Error downloading a wallet statement"?

**Comment by JmillsExpensify** [comment]
Oh cool! That's awesome, I don't personally think removing that is an issue since the wallet statement is still a beta, though @joekaufmanexpensify can jump in and confirm one way or another.

**Comment by joekaufmanexpensify** [comment]
Just for my own understanding, would we be replacing the growl with a different style of error message? Or just removing the growl and leave it failing silently for now? 

I think either way is fine for now (since no one is really using the statement right now). Although if we aren't adding a new error now, I'll add this to the [existing statements cleanup issue](https://github.com/Expensify/Expensify/issues/213310) we have (which is on hold right now).

**Comment by yuwenmemon** [comment]
Removing the HOLD from this!

**Comment by JmillsExpensify** [comment]
@joekaufmanexpensify Yeah we'd replace the growl with a different message style.

**Comment by joekaufmanexpensify** [comment]
Sounds good! Yeah if we're replacing the growl with a different error message style, that sounds great.

**Comment by joekaufmanexpensify** [comment]
Just for my own understanding, have we decided what will be added in their place?

**Comment by luacmartins** [comment]
IMO we should use the existing `DotIndicatorMessage` and `Form` error pattern when possible:

For example, here's a case where the modal was replaced with the `DotIndicatorMessage` component:

![Screenshot 2023-01-25 at 10 34 27 AM](https://user-images.githubusercontent.com/22219519/214639134-e143daf4-1e18-4871-a515-8c40b31fd5ad.png)


And here's how we show it in Form:

![Screenshot 2023-01-25 at 10 34 41 AM](https://user-images.githubusercontent.com/22219519/214639194-895810ae-1297-4b81-b928-8ec76716724e.png)

**Comment by JmillsExpensify** [comment]
Agree with @luacmartins

**Comment by joekaufmanexpensify** [comment]
Makes sense, ty!

**Comment by kadiealexander** [comment]
Is adding in the new indicator messages something I should do as a part of this issue, or will it be covered later? Just clarifying because the scope of the OP was very limited to removing growls.

**Comment by JmillsExpensify** [comment]
Sorry what's the new indicator messages?

**Comment by luacmartins** [comment]
I _think_ @kadiealexander is talking about the new error messages we'd put in place instead of the growl?

**Comment by JmillsExpensify** [comment]
Gotcha, hmm yeah I think we need the error messages to appear somewhere, if we are deprecating the growls where they appear today, right? Apologies if I'm missing something obvious.

**Comment by luacmartins** [comment]
Yea, I agree

**Comment by kadiealexander** [comment]
Okay! The OP is very limited to removing the current growls, thanks for the added context. I'm going to pause on actioning this while I figure out the new messaging for these. A few aren't errors, so I'm not sure if any sort of "success" message should be put in place instead.

Not an error, need to determine what message to show (if any):

- Interacting with the checkbox in Workspace > Manage member page
- Adding/Deleting PaypalMe address (not an error)

Growl errors to be replaced with `DotIndicatorMessage`:

- Failure updating the User avatar
- Failure to create, update or delete a Workspace
- Error downloading a wallet statement

**Comment by luacmartins** [comment]
>Interacting with the checkbox in Workspace > Manage member page

I believe this one was done as part of [this PR](https://github.com/Expensify/App/pull/14347) already. 

For the non-error ones, we've been implementing them like [this](https://github.com/Expensify/App/blob/5937ed9fc12455b7f81007e1c495af46193e8e02/src/pages/signin/ResendValidationForm.js#L75-L76)

**Comment by kadiealexander** [comment]
Thanks @luacmartins! Here's where I'm at after some fishing around, would love some opinions on it.

Interacting with the checkbox in Workspace > Manage member page - Fixed (I saw that while testing just now as well)
Adding/Deleting PaypalMe address (not an error) - Fix with success dot indicator as per [this](https://github.com/Expensify/App/blob/5937ed9fc12455b7f81007e1c495af46193e8e02/src/pages/signin/ResendValidationForm.js#L75-L76)
Failure updating the User avatar - I actually think this is fine as-is, it replicates our other upload errors:

User avatar:

![image](https://user-images.githubusercontent.com/59587260/215912235-32f0b95b-68c3-40ce-aa59-b5448ac91cbe.png)

Other media upload error:

![image](https://user-images.githubusercontent.com/59587260/215912266-fe6008da-e99a-4571-830b-5e961715316b.png)

Failure to create, update or delete a Workspace - I can't replicate this (can't seem to break a workspace which is good!) but I'll remove any growls and replace with failure indicators. 

Error downloading a wallet statement - As discussed above, I'll remove any growls and replace with failure indicators.

**Comment by luacmartins** [comment]
Yea, I think `Failure updating the User avatar` is fine as is! Everything else sounds good to me too.

**Comment by luacmartins** [comment]
>Failure to create, update or delete a Workspace

I haven't tried this, but it might take you in the direction to trigger an error:

1. Open the App in two browser windows, A and B in incognito
2. Log in to the same account
3. In window A, go offline
4. In window B, delete the workspace
5. In window A, edit the workspace name
6. In window A, come back online

**Comment by kadiealexander** [comment]
Our engineers are too good, everything behaved as expected (in window A during step 6 the workspace just disappeared).

**Comment by kadiealexander** [comment]
Ah, awesome! Thanks @luacmartins. In that case, the workspace setting errors all seem to be fine. I'll still surf through and see if there are any old-style growls left, but it's looking promising on that front.

**Comment by conorpendergrast** [comment]
@kadiealexander Got a summary of what's done, and what's left to do? If so, I'd be happy to take this back over!

**Comment by kadiealexander** [comment]
[This](https://github.com/Expensify/App/issues/13229#issuecomment-1411265468) still applies. When we started talking about replacing the growls with new indicators the complexity rose above my skill level with ReactNative so I'm working this through with my mentor @yuwenmemon, and would love to keep it if that's okay! We have a sync tomorrow to talk through how to replace the growls.

**Comment by conorpendergrast** [comment]
Fantastic, go for it Kadie. No pressing urgency from what I understand, and a steady pace while you learn sounds great!

**Comment by conorpendergrast** [comment]
Looks like this is ongoing still @kadiealexander, is that right? No rush at all on my end

**Comment by JmillsExpensify** [comment]
Agree that there is no rush, though is would still be good to do this in the intermediate term. I *think* we don't have many growls as it is.

**Comment by kadiealexander** [comment]
I've discussed this with my mentor @puneetlath and we agreed that the addition of different indicators blew this out of my realm of expertise. I'm unassigning so that this is not blocked on me.

**Comment by conorpendergrast** [comment]
@Puneet-here @kadiealexander This still needs to be internal, correct?

**Comment by kadiealexander** [comment]
Wrong Puneet! From everything I saw this could be external, but I could be wrong.

**Comment by conorpendergrast** [comment]
Oh that's embarrassing, silly me

**Comment by puneetlath** [comment]
Agreed, seems like it can definitely be external. If it's a little more involved, it could be a good one for an expert agency.
